src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -873,26 +920,20 @@
   [{:keys [projections]}]
   (->> projections
        (remove (comp :query-only? val))
```

Comment:
You changed this call to `(comp :query-only? second)` below, might want to change this one as well, but I actually like using `val` instead since it signals that the args are actually map kv pairs.